src/conn/conn_layered.c

```diff
@@ -1097,14 +1097,14 @@ __wt_conn_is_disagg(WT_SESSION_IMPL *session)
  *     Delete a file or fail depending on the flag.
  */
 static int
-__disagg_delete_or_fail(WT_SESSION_IMPL *session, const char *fname, bool fail)
+__disagg_delete_or_fail(WT_SESSION_IMPL *session, const char *fname, bool fail, bool ok_not_exist)
 {
-    bool file_exists;
-
-    WT_RET(__wt_fs_exist(session, fname, &file_exists));
-
-    if (!file_exists)
-        return (0); /* Nothing to do, file does not exist. */
+    if (ok_not_exist) {
+        bool file_exists;
+        WT_RET(__wt_fs_exist(session, fname, &file_exists));
+        if (!file_exists)
+            return (0); /* Nothing to do, file does not exist. */
+    }
 
     if (fail)
```

Comment:
Does this work properly now? If we have `fail` and not `ok_not_exist`, we'll just immediately return `EEXIST` here. I'll grant that combination is weird, but it makes me wonder if there's a simpler version of the function possible.

Reply:
AFAIK, `__wt_fs_exist()` returns 0 on successful completion, with the file's existence status provided in the output variable passed as an argument. It doesn't use the return value itself to indicate whether the file exists.